### PR TITLE
TDD: Creating solver and tests 

### PR DESCRIPTION
**Pull request by hernandanielzamora**
## In this PR the following changes have been made:

- Created solver file with it methods:
    - Factorial
    - Reverse
    - Fizzbuzz
- Created solver_spec file that contains tests for the previous methods. 
- This project was built using TDD (wrote the tests first and then the code)

### CODE REVIEW

File: solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  describe '#factorial' do
+    it 'Returns 1 when the inputed number is 0' do
+      expect(solver.factorial(0)).to eq(1)
+    end
+
+    it 'Returns the factorial of a positive integer' do
+      expect(solver.factorial(8)).to eq(40_320)
+      expect(solver.factorial(4)).to eq(24)
+    end
```
Comment (tresorsawasawa) [inline solver_spec.rb:14]: - [ ] As required for the project, kindly ensure that the `#factorial` method also has tests which handle `Negative value which raises an exception as an expectation.`

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @hernandanielzamora, @sameedmohsin28, and @dazehinn 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job implementing the changes requested by the previous CR.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
